JS/index.js

```diff
@@ -200,23 +216,27 @@ function isGameOver() {
 async function createWaves() {
     for (let i = 0; GAME_STATE === GAME_STATE_ACTIVE && i < 10; i++) {
         for (let times = 0; GAME_STATE === GAME_STATE_ACTIVE && times < 4; times++) {
-            await addEnemy(times);
             if (isGameOver()) {
                 return;
             }
+            let enemy = addEnemy(times);
+            moveEnemyDown(enemy);
+            await new Promise(r => setTimeout(r, waves));
         }
         for (let timez = 3; GAME_STATE === GAME_STATE_ACTIVE && timez >= 0; timez--) {
-            await addEnemy(timez);
-            if (GAME_STATE === GAME_STATE_OVER) {
+            if (isGameOver()) {
                 return;
             }
+            let enemy = addEnemy(timez);
+            moveEnemyDown(enemy);
+            await new Promise(r => setTimeout(r, waves));
         }
     }
 }
 
 async function createDoubleWaves() {
     if (GAME_STATE === GAME_STATE_ACTIVE) {

```

Comment:
You validate game state twice here and this makes no sense. Remove external "if"

Reply:
fin